src/test/java/bio/terra/integration/DataRepoClient.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.integration;
+
+import bio.terra.model.DeleteResponseModel;
+import bio.terra.model.ErrorModel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * This class holds a Spring RestTemplate
+ */
+@Component
+public class DataRepoClient {
+    @Value("${integrationtest.port}")
+    private String testPort;
+
+    @Value("${integrationtest.server}")
+    private String testServer;
+
+    @Value("${integrationtest.protocol}")
+    private String testProtocol;
+
+    private RestTemplate restTemplate;
+    private ObjectMapper objectMapper;
+    private HttpHeaders headers;
+
+    public DataRepoClient() {
+        restTemplate = new RestTemplate();
+        restTemplate.setErrorHandler(new DataRepoClientErrorHandler());
+        objectMapper = new ObjectMapper();
+
+        headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON_UTF8);
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON_UTF8));
+    }
+
```

Comment:
most of these 4 methods are the same. you could make an internal makeRequest method that took the entity and and the HttpMethod and you could reduce much of the duplicated code. the individual methods would basically be creating the correct entity...

Reply:
Good idea. There started out more different, but when I had to switch to the exchange call them ended up more the same.